infra/lib/attendant-ide/index.ts

```diff
@@ -0,0 +1,250 @@
+    super(scope, id);
+
+    // const defaultVpc = Vpc.fromLookup(this, 'defaultVpc', { isDefault: true });
+    const vpc = new Vpc(this, 'VPC', {
+      maxAzs: 2,
+      subnetConfiguration: [
+        {
+          cidrMask: 24,
+          name: 'Public',
+          subnetType: SubnetType.PUBLIC,
+        },
+        {
+          cidrMask: 24,
+          name: 'Private',
+          subnetType: SubnetType.PRIVATE_WITH_EGRESS,
+        },
+      ],
+      flowLogs: {
+        VPCFlowLogs: {
+          destination: FlowLogDestination.toCloudWatchLogs(),
+          trafficType: FlowLogTrafficType.REJECT,
+        },
+      },
+    });
+    const instanceSecurityGroup = new SecurityGroup(this, 'instance-sg', {
+      vpc,
+      allowAllOutbound: true,
+    });
+
+    const idePasswordParameter = new StringParameter(this, 'ide-password', {
+      dataType: ParameterDataType.TEXT,
+      stringValue: 'placeholder-changed-at-deploy-time-by-ec2-user-data',
+      parameterName: 'vscode-password',
+    });
+
+    const userData = UserData.forLinux();
+    userData.addCommands(
+      // Install general dependencies
+      'yum install -y git docker jq zsh util-linux-user gcc make gcc-c++ libunwind java-17-amazon-corretto-headless unzip zip zlib zlib-devel openssl-devel ncurses-devel readline-devel bzip2-devel libffi-devel sqlite-devel xz-devel',
+      // Setup docker
+      'service docker start',
+      'usermod -aG docker ec2-user',
+      // Setup zsh and oh-my-zsh
+      'chsh -s $(which zsh) ec2-user',
+      `su - ec2-user -c 'sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)" "" --unattended'`,
+      // Download .zshrc
+      // TODO: change to main branch once merged
+      `su - ec2-user -c 'curl -fsSL https://raw.githubusercontent.com/aws-samples/powertools-for-aws-lambda-workshop/ec2-ide/infra/lib/attendant-ide/zshrc-sample.txt -o $HOME/.zshrc'`,
+      // Install Pyenv
+      `su - ec2-user -c 'curl -s https://pyenv.run | zsh'`,
+      // Install fnm (Node.js)
+      "su - ec2-user -c 'curl -fsSL https://fnm.vercel.app/install | zsh'",
+      // Setup Python
+      `su - ec2-user -c 'export PATH="/home/ec2-user/.pyenv/bin:$PATH" && eval "$(pyenv init --path)" && pyenv install 3.11.0'`,
+      `su - ec2-user -c 'export PATH="/home/ec2-user/.pyenv/bin:$PATH" && eval "$(pyenv init --path)" && pyenv global 3.11.0'`,
+      // Setup Node
+      `su - ec2-user -c 'export PATH="/home/ec2-user/.local/share/fnm:$PATH" && eval "\`fnm env\`" && fnm install v18'`,
+      `su - ec2-user -c 'exso port PATH="/home/ec2-user/.local/share/fnm:$PATH" && eval "\`fnm env\`" && fnm default v18'`,
+      `su - ec2-user -c 'export PATH="/home/ec2-user/.local/share/fnm:$PATH" && eval "\`fnm env\`" && fnm use v18'`,
+      // Install .NET
+      'rpm -Uvh https://packages.microsoft.com/config/centos/7/packages-microsoft-prod.rpm',
+      'yum install -y aspnetcore-runtime-6.0 dotnet-sdk-6.0',
```

Comment:
After installing .NET we can also add Lambda dotnet tools to the cli.

`dotnet tool install -g Amazon.Lambda.Tools`
`dotnet new -i "Amazon.Lambda.Templates::*"`

More info: https://github.com/aws/aws-lambda-dotnet#amazonlambdatools 

Package adds commands to the dotnet cli that can be used to manage Lambda functions including deploying a function from the dotnet cli. For more information see the [README.md](https://github.com/aws/aws-lambda-dotnet/blob/master/Libraries/src/Amazon.Lambda.Tools/README.md) file for Amazon.Lambda.Tools.

Reply:
Added, thanks!